src/psyclone/psyir/symbols/symboltable.py

```diff
@@ -913,6 +931,20 @@ def tags_dict(self):
         '''
         return self._tags
 
+    def get_reverse_tags_dict(self):
+        '''
+        Constructs and returns a reverse of the map returned by
+        :py:method:`tags_dict`.
+
+        :returns: ordered dictionary of tags indexed by symbol.
+        :rtype: OrderedDict[:py:class:`psyclone.psyir.symbols.Symbol`] = str
+
+        '''
+        tags_dict_reversed = OrderedDict()
+        for tag, sym in self._tags.items():
+            tags_dict_reversed[sym] = tag
```

Comment:
Oh yes, that did sort of float across my mind while I was doing this and then I forgot about it. I've extended the code and will add a test.

Reply:
In fact, implementing support for this means reworking other bits of the API so that a list of tags can be specified. I'm therefore going to suggest leaving this for now and have created a new Issue for it - #1654.